scripts/base/init-bare.zeek

```diff
@@ -5286,6 +5286,10 @@ export {
 	## interfaces.
 	const bufsize = 128 &redef;
 
+	## Number of bytes to use for buffering PCAP file read operations.
+	## Setting this to 0 uses operating system defaults (likely 4096).
```

Comment:
I feel like this needs to do more explaining on how to arrive at a locally good value (and TBH I am not convinced the new default couldn't lead to perf regressions in typical setups). Things to take into account are probably fs block size and the CPU cache sizes.

Reply:
Hmm, if this causes a regression we learned something and it can be tweaked. IMO this is not at a level where CPU cache size should be relevant. The main effect is reduction the number of `read()` system calls by a factor of 32 for large pcaps in usual configurations (Linux + default ext4) at a minor increase of memory usage (128kb in 2023).

As a compromise, could default to `0`, but then I can almost keep it as a private patch for my own benefit :-)

Reply:
@J-Gras chimed in mentioning he investigated a similar optimization for a PCAP load-balancer/splitter. He also provided a link to coreutils code that also uses 128kb today by default:

https://github.com/coreutils/coreutils/blob/9581c4b59d2e11cd090613cd300a833b4441d2b5/src/ioblksize.h#L24-L58

Reply:
Based on the script in that code, going even larger is better on an M1:

```
tim@coregeek2 build% for i in $(seq 0 10); do
     bs=$((1024*2**$i))
     printf "%7s=" $bs
     timeout --foreground -sINT 2 \
       gdd bs=$bs if=/dev/zero of=/dev/null 2>&1 \
         | sed -n 's/.* \([0-9.]* [GM]B\/s\)/\1/p'
   done
   1024=1.2 GB/s
   2048=2.4 GB/s
   4096=4.3 GB/s
   8192=8.0 GB/s
  16384=13.8 GB/s
  32768=20.8 GB/s
  65536=28.8 GB/s
 131072=34.9 GB/s
 262144=39.6 GB/s
 524288=41.7 GB/s
1048576=43.7 GB/s
```

I agree with the argument for going to a larger default though. The current default of 4k is likely too small for modern systems.